Shared/Samples/Search for web map/SearchForWebMapView.swift

```diff
@@ -0,0 +1,180 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct SearchForWebMapView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The text query in the search bar.
+    @State private var query = ""
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        ZStack {
+            ScrollView {
+                LazyVStack {
+                    ForEach(model.portalItems, id: \.id) { item in
+                        NavigationLink {
+                            SafeMapView(map: Map(item: item))
+                                .navigationTitle(item.title)
+                        } label: {
+                            PortalItemRowView(item: item)
+                        }
+                        .buttonStyle(.plain)
+                        .onAppear {
+                            // Load the next results when the last item is reached.
+                            if item.id == model.portalItems.last?.id {
+                                do {
+                                    try model.findNextItems()
+                                } catch {
+                                    self.error = error
+                                }
+                            }
+                        }
+                    }
+                }
+                
+                ProgressView()
+                    .padding()
+                    .opacity(model.isLoadingResults ? 1 : 0)
+            }
+            
+            VStack {
+                Text("No Results")
+                    .font(.headline)
+                Text("Check sspelling or try a new search.")
+                    .font(.footnote)
+            }
+            .opacity(!model.isLoadingResults && !query.isEmpty && model.portalItems.isEmpty ? 1 : 0)
+        }
+        .background(Color(.secondarySystemBackground))
+        .searchable(
+            text: $query,
+            placement: .navigationBarDrawer(displayMode: .always),
+            prompt: "Search for a web map"
+        )
+        .onChange(of: query) { _ in
```

Comment:
Change this to a `task(id:priority:_:)` modifier. If the id value changes, SwiftUI cancels and restarts the task.